e2e/commands_test.go

```diff
@@ -184,71 +184,72 @@ func TestSplitMerge(t *testing.T) {
 	tmpDir := fs.NewDir(t, "split_merge")
 	defer tmpDir.Remove()
 
-	icmd.RunCommand(dockerApp, "merge", "testdata/render/envvariables/my.dockerapp", "-o", tmpDir.Join("remerged.dockerapp")).Assert(t, icmd.Success)
+	cmd := icmd.Cmd{Command: dockerCli.Command("app", "merge", "testdata/render/envvariables/my.dockerapp", "--output", tmpDir.Join("remerged.dockerapp"))}
+	icmd.RunCmd(cmd).Assert(t, icmd.Success)
 
-	cmd := icmd.Cmd{Dir: tmpDir.Path()}
+	cmd.Dir = tmpDir.Path()
 
 	// test that inspect works on single-file
-	cmd.Command = []string{dockerApp, "inspect", "remerged"}
+	cmd.Command = dockerCli.Command("app", "inspect", "remerged")
 	result := icmd.RunCmd(cmd).Assert(t, icmd.Success)
 	golden.Assert(t, result.Combined(), "envvariables-inspect.golden")
 
 	// split it
-	cmd.Command = []string{dockerApp, "split", "remerged", "-o", "split.dockerapp"}
+	cmd.Command = dockerCli.Command("app", "split", "remerged", "--output", "split.dockerapp")
 	icmd.RunCmd(cmd).Assert(t, icmd.Success)
 
-	cmd.Command = []string{dockerApp, "inspect", "remerged"}
+	cmd.Command = dockerCli.Command("app", "inspect", "remerged")
 	result = icmd.RunCmd(cmd).Assert(t, icmd.Success)
 	golden.Assert(t, result.Combined(), "envvariables-inspect.golden")
 
 	// test inplace
-	cmd.Command = []string{dockerApp, "merge", "split"}
+	cmd.Command = dockerCli.Command("app", "merge", "split")
 	icmd.RunCmd(cmd).Assert(t, icmd.Success)
 
-	cmd.Command = []string{dockerApp, "split", "split"}
+	cmd.Command = dockerCli.Command("app", "split", "split")
 	icmd.RunCmd(cmd).Assert(t, icmd.Success)
 }
 
 func TestBundle(t *testing.T) {
 	tmpDir := fs.NewDir(t, t.Name())
 	defer tmpDir.Remove()
 	// Using a custom DOCKER_CONFIG to store contexts in a temporary directory
-	cmd := icmd.Cmd{Env: append(os.Environ(), "DOCKER_CONFIG="+tmpDir.Path())}
+	cmd := icmd.Cmd{Env: os.Environ()}
```

Comment:
need to patch DOCKER_CONFIG here

Reply:
This has been reworked so that we use a new config dir for each test.

We can actually remove the `Env: os.Environ()` as it's redundant. Let's do so in a followup @ulyssessouza

Reply:
Looked at the new dir for each test thing. The main problem is that it changes the environment for the whole test project, while only the child processes we invoke require these environment variables changes. Also, this dockerCli global variable makes it impossible to run tests in parallel.